packages/venia-concept/src/components/MiniCart/miniCart.js

```diff
@@ -26,12 +29,35 @@ const MiniCart = props => {
     } = props;
     const { editItem, isEditingItem, isLoading, isUpdatingItem } = cart;
 
+    // Hooks.
+    const { innerHeight } = useWindowSize();
+    const [viewportHeight, setViewportHeight] = useState(innerHeight);
+    // We need innerHeight to be mutable so its value is not captured in
+    // our debounced function.
+    const innerHeightMutableRef = useRef(innerHeight);
+
+    // Always update the innerHeight mutable ref with innerHeight's latest value.
+    innerHeightMutableRef.current = innerHeight;
+
+    const setViewportHeightDebounced = debounce(() => {
+        // We're really just doing setViewportHeight(innerHeight) here
+        // but because we're within a debounced function, innerHeight
+        // gets captured (never updates).
+        // So we use this ref instead to ensure we always have the latest value.
+        setViewportHeight(innerHeightMutableRef.current);
+    }, 1000);
+    const setViewportHeightDebouncedRef = useRef(setViewportHeightDebounced);
```

Comment:
I didn't add a comment in the code here but just FYI we also had to make the debounced function a ref because otherwise a new debounced function would get created each time through this function and they would ALL fire after the debounce delay.

Reply:
Can we not just `useMemo` here?

Reply:
Good question - I had that thought too but for some reason tried `useRef` first. Since it worked I didn't try `useMemo`.

If you think it's better I can switch it.

Reply:
Isn't this also going to multiple references to `setViewportHeightDebounced` and only invoke one? As in whichever is attached to the `ref` last? It just feels overly confusing currently but I can see now how my original example would have queued a bunch of `useEffect` callbacks too...